approaches/byte-pair-encoding.md

```diff
@@ -5,3 +5,25 @@ layout: coming_soon
 title: Byte-pair encoding
 description: A data compression algorithm for machine translation
 ---
+
+Byte-pair encoding (BPE) is a data compression algorithm where the most common consecutive pair of bytes is replaced by a byte that is not seen in the data.
+It was first introduced in the article titled "A new Algorithm for Data Compression" by Philip Gage in the February 1994 edition of C Users Journal.
+
+A very well-known example of BPE data compression is, suppose we have a string "aaabdaaabac" which is compressed such that, the most frequent byte pair "aa" is replaced by Z. We now have "ZabdZabac". 
+The new most frequent byte pair is "ab" so we replace it by Y such that, we now have "ZYdZYac". 
+Applying recursive byte pair encoding to encode ZY as X, we now have "XdXac". It cannot be compressed further and to decompress the data, replacement in reverse order is performed. 
+
+The use of BPE in NLP ensures that the most common words in the vocabulary are presented as single token whereas the rarest words are broken down into subword tokens instead of introducing it as `<UNK>` tokens.
```

Comment:
Although it is understood, please [expand the acronym that hasn't been explained in the article](https://machinetranslate.org/style#acronyms-and-abbreviations). In this case, NLP.